op-dispute-mon/mon/detector_test.go

```diff
@@ -200,22 +208,33 @@ func TestDetector_CheckAgreement_Succeeds(t *testing.T) {
 	for _, test := range tests {
 		test := test
 		t.Run(test.name, func(t *testing.T) {
-			detector, _, _, _ := setupDetectorTest(t)
+			detector, _, _, _, logs := setupDetectorTest(t)
 			batch, err := detector.checkAgreement(context.Background(), common.Address{}, 0, test.rootClaim, test.status)
 			require.NoError(t, err)
 			test.expectBatch(&batch)
+
+			if !test.expectErrorLog {
+				require.Empty(t, logs.FindLogsWithLevel(log.LevelError), "Should not log an error`")
+			} else {
+				l := logs.FindLog(log.LevelError, "Unexpected game result")
+				require.NotNil(t, l, "Should have logged an error")
+				expectedResult := l.AttrValue("expectedResult")
+				require.Equal(t, test.expectStatus, expectedResult)
+				actualResult := l.AttrValue("actualResult")
+				require.Equal(t, test.status, actualResult)
+			}
 		})
 	}
 }
 
-func setupDetectorTest(t *testing.T) (*detector, *mockDetectorMetricer, *mockMetadataCreator, *stubOutputValidator) {
-	logger := testlog.Logger(t, log.LvlDebug)
+func setupDetectorTest(t *testing.T) (*detector, *mockDetectorMetricer, *mockMetadataCreator, *stubOutputValidator, *testlog.CapturingHandler) {
+	logger, capturedLogs := testlog.CaptureLogger(t, log.LvlDebug)
```

Comment:
This is very cool, good to know. Will use this for the forecaster to expect logging of the game resolution status.